test/integration/ClientEncryptionTest.java

```diff
@@ -0,0 +1,138 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.vaticle.typedb.client.test.integration;
+
+import com.vaticle.typedb.client.TypeDB;
+import com.vaticle.typedb.client.api.TypeDBClient;
+import com.vaticle.typedb.client.api.TypeDBCredential;
+import com.vaticle.typedb.client.common.exception.TypeDBClientException;
+import com.vaticle.typedb.common.test.cluster.TypeDBClusterRunner;
+import org.junit.After;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.vaticle.typedb.client.common.exception.ErrorMessage.Client.CLUSTER_CONNECTION_CLOSED_UKNOWN;
+import static com.vaticle.typedb.client.common.exception.ErrorMessage.Client.CLUSTER_ENDPOINT_NOT_ENCRYPTED;
+import static com.vaticle.typedb.client.common.exception.ErrorMessage.Client.CLUSTER_SSL_CERTIFICATE_NOT_VALIDATED;
+import static com.vaticle.typedb.client.common.exception.ErrorMessage.Client.CLUSTER_UNABLE_TO_CONNECT;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+public class ClientEncryptionTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ClientEncryptionTest.class);
+    private static final String USERNAME = "admin";
+    private static final String PASSWORD = "password";
+
+    private static final Path WRONG_CA = Path.of("test/integration/resources/wrong-root-ca.pem");
+    private static final Path CLUSTER_CERTIFICATE = Path.of("test/integration/resources/dev-certificate.pem");
+    private static final Path CLUSTER_PRIVATE_KEY = Path.of("test/integration/resources/dev-private-key.pem");
+    private static final Path CLUSTER_CA = Path.of("test/integration/resources/dev-root-ca.pem");
+    private static final Map<String, String> CLUSTER_ENCRYPTION_OPTIONS = Map.ofEntries(
+            Map.entry("--server.encryption.enable", "true"),
+            Map.entry("--server.encryption.certificate", CLUSTER_CERTIFICATE.toAbsolutePath().toString()),
+            Map.entry("--server.encryption.private-key", CLUSTER_PRIVATE_KEY.toAbsolutePath().toString()),
+            Map.entry("--server.encryption.root-ca", CLUSTER_CA.toAbsolutePath().toString())
+    );
+
+
+    private static final TypeDBCredential ENCRYPTED_CREDENTIALS = new TypeDBCredential(USERNAME, PASSWORD, CLUSTER_CA);
+    private static final TypeDBCredential MISCONFIGURED_CREDENTIALS = new TypeDBCredential(USERNAME, PASSWORD, WRONG_CA);
+    private static final TypeDBCredential UNENCRYPTED_CREDENTIALS = new TypeDBCredential(USERNAME, PASSWORD, false);
+
+    private TypeDBClusterRunner clusterRunner;
+
+    @After
+    public void shutDownCluster() {
+        if (clusterRunner != null) {
+            clusterRunner.stop();
+            clusterRunner.deleteFiles();
```

Comment:
Check if this is already included in the cluster runner cleanup

Reply:
There's no cleanup.
Just a `reset` or a `deleteFiles`
This feels more explicit.